### ISSUE TITLE
[Bug]: Portfolio - the balance is not updated correctly when user switches source tokens

### ISSUE THREAD

**Issue by sleepytanya** [MetaMask/metamask-extension#25620]
### Describe the bug

    During the swapping process: 
- when a user switches the source token, the balance doesn't automatically reset to zero
-  If the balance was previously set to 'Max', after getting quotes and rejecting the tx, it defaults back to the balance of the native token. 
    Consequently, this triggers an 'Insufficient balance' warning which can be confusing to users.

### Expected behavior

Token balances reflected correctly.

### Screenshots/Recordings

https://github.com/MetaMask/metamask-extension/assets/104780023/7336a8de-a9e7-434a-9ace-bc63e1c667c1

### Steps to reproduce

1. Go to MM Portfolio
2. Start Swap tx
3. Set some amount for the source token
4. Switch source token
5. Notice that the amount is not reset to 0 
6. Set 'Max' amount
7. Switch source token, get quotes and reject the tx
8. Notice that 'Max' amount for the selected token is replaced with the 'Max' amount for the native token

### Error messages or log output

```shell
No errors
```

### Version

11.16.14

### Build type

None

### Browser

Chrome

### Operating system

MacOS

**Comment by gauthierpetetin** [comment]
Hi @sleepytanya , as this bug applies to Portfolio, shall it be moved to Portfolio's bug tracking system? (i.e. in Jira I believe?)

**Comment by sleepytanya** [comment]
@gauthierpetetin  Yes, you are right! But I know nothing about Portfolio's bug tracking and I don''t think I have an access to it. If you don't have an access as well we can just let them know and close this ticket?

**Comment by darkwing** [comment]
Closed, moving to Portfolio repo.
